lsp-julia.el

```diff
@@ -25,33 +30,35 @@
 
 (defcustom lsp-julia-timeout 30
   "Time before lsp-mode should assume julia just ain't gonna start."
+  :type 'number
   :group 'lsp-julia)
 
 (defun lsp-julia--get-root ()
-  "Try to find the package directory by searching for a .gitignore file.
-If no .gitignore file can be found use the default directory "
-  (let ((dir (locate-dominating-file default-directory ".gitignore")))
-    (if dir
-        (expand-file-name dir)
-      default-directory)))
+  (let ((dir (locate-dominating-file default-directory "Project.toml")))
+    (if dir (expand-file-name dir)
+      (expand-file-name lsp-julia-default-environment))))
 
 (defun lsp-julia--rls-command ()
-  `(,lsp-julia-command ,@lsp-julia-flags "-e using LanguageServer; server = LanguageServer.LanguageServerInstance(STDIN, STDOUT, false); server.runlinter = true; run(server);"))
-
+  `(,lsp-julia-command
+    ,@lsp-julia-flags
+    ,(concat "-e using LanguageServer, Sockets, SymbolServer;"
+             " server = LanguageServer.LanguageServerInstance("
+             " stdin, stdout, false,"
+             " \"" (lsp-julia--get-root) "\","
+             " \"\", Dict());"
+             " server.runlinter = true;"
+             " run(server);")))
 
 (defconst lsp-julia--handlers
   '(("window/setStatusBusy" .
      (lambda (w _p)))
     ("window/setStatusReady" .
      (lambda(w _p)))))
 
-(defun lsp-julia--initialize-client(client)
-  (mapcar #'(lambda (p) (lsp-client-on-notification client (car p) (cdr p))) lsp-julia--handlers)
-  (setq-local lsp-response-timeout lsp-julia-timeout))
-
-(lsp-define-stdio-client lsp-julia "julia" #'lsp-julia--get-root nil
-                         :command-fn #'lsp-julia--rls-command
-                         :initialize #'lsp-julia--initialize-client)
+(lsp-register-client
+ (make-lsp-client :new-connection (lsp-stdio-connection (lsp-julia--rls-command))
```

Comment:
it is better to change  `(lsp-julia--rls-command)` to  `'lsp-julia--rls-command` to force recalculation of Julia server params each time new server is started.

Reply:
done